conans/client/rest/file_downloader.py

```diff
@@ -86,24 +95,43 @@ def write_chunks(chunks, path):
                 ret = bytes(ret_data)
             return ret, downloaded_size
 
+        def get_total_length():
+            content_range = response.headers.get("Content-Range", "")
+            match = re.match(r"^bytes (\d+)-(\d+)/(\d+)", content_range)
+            if match:
+                assert range_start == int(match.group(1))
```

Comment:
This could be a failure in the protocol or server side, maybe launch a ``ConanException`` with an informative message of what happened? That might help us to support if some user report errors.

Reply:
There is a use case where this method return ``None``, which  will produce confusing NoneType exception later, if there is no match.